client/src/components/InternshipTag.jsx

```diff
@@ -1,25 +1,29 @@
-import { Box } from '@mui/material';
+import { Box, Grid } from '@mui/material';
 import React from 'react';
 
 const InternshipTag = (props) => {
-  const Icon = props.icon;
+  const icon = props.icon;
 
   return (
-    <Box
-      display="flex"
-      maxWidth={false}
-      sx={{
-        borderRadius: '0.5rem',
-        padding: '0.5rem 1rem',
-        bgcolor: 'secondary.light',
-        border: 1,
-        borderColor: 'text.main',
-        maxWidth: 'fit-content',
-      }}
-    >
-      <Icon />
-      <Box sx={{ padding: '0rem 0.25rem' }}>{props.label}</Box>
-    </Box>
+    <Grid item>
```

Comment:
I think having the Grid item tag in the return of InternshipCard.jsx makes more sense, since the context of a grid container is there.